File: perl/filereader.pl
#!/usr/bin/perl
use strict;
use warnings;
use Getopt::Std;
our ($opt_f, $opt_n, $opt_c);
my ($line);
getopt('f:n:c:');
die "Please enter a file to process using -f <filename>\n" unless (defined $opt_f);
die "Please enter a new name using -n <name>\n" unless (defined $opt_n);
die "Please enter the name to change using -c <name>\n" unless (defined $opt_c);
die "Unable to open input file " unless (open my $infile, "<", $opt_f);
die "Unable to open output file " unless (open my $oufile, ">", $opt_f . ".out");
while (<$infile>) {
    chomp;
    $line = $_;
    $line =~ s/$opt_c/$opt_n/g;
    print $oufile $line . "\n";
}
close $infile;
close $oufile;

File: php/filereader.php
<?php
   $filename = '';
   $name = '';
   $fromname = '';
   $nextf = 0;
   $nextn = 0;
   $nextc = 0;
   foreach ($argv as $arg) {
      if ( $nextf == 1 ) {
         $filename = $arg;
         $nextf = 0;
      }
      if ( $arg == "-f" ) {
         $nextf = 1;
      }
      if ( $nextn == 1 ) {
         $name = $arg;
         $nextn = 0;
      }
      if ( $arg == "-n" ) {
         $nextn = 1;
      }
      if ( $nextc == 1 ) {
         $fromname = $arg;
         $nextc = 0;
      }
      if ( $arg == "-c" ) {
         $nextc = 1;
      }
   }
   echo "Processing " . $filename . "\n";
   $infile = fopen($filename, "r") or die ("Unable to open input file!");
   $outfile = fopen($filename . ".out", "w") or die ("Unable to open output file!");
   while (!feof($infile)) {
      $line = fgets($infile);
      fwrite($outfile, str_replace($fromname, $name, $line));
   }
   fclose($infile);
   fclose($outfile);
?>
